### PR TITLE
add console.group & console.groupEnd

### PR DESCRIPTION
**Pull request by zhmushan**

### CODE REVIEW

File: js/console.ts [hunk 1]
```diff
@@ -473,4 +475,15 @@ export class Console {
 
     this.info(`${label}: ${duration}ms`);
   };
+
+  group = (...label: string[]): void => {
```
Comment (justjavac) [inline js/console.ts]: see [WebIDL](https://console.spec.whatwg.org/#idl-index), use `group = (...label: any[])` instead

File: js/console.ts [hunk 2]
```diff
@@ -346,14 +346,16 @@ export function stringifyArgs(
       );
     }
   }
-  return out.join(" ");
+  return `${groupIndent}${out.join(" ")}`;
```
Comment (justjavac) [inline js/console.ts]: If the `log` message contains `"\n"`, will be output to the wrong location.

```ts
console.group('g')
console.log("foo\nbar")
console.groupEnd();
```

output:

```
g
  foo
bar
```

Comment (zhmushan) [reply js/console.ts]: I think I need to consider how to fix it.

**Comment by hayd** [comment]
Would it be weird to special case a group label to do the single line, e.g. `group("")`?

I think that change is a little more complicated (requires modifying printFunc) so it could be done later...

**Comment by justjavac** [comment]
what about `groupCollapsed` https://github.com/denoland/deno/pull/1355#issuecomment-448109488

**Comment by zhmushan** [comment]
@justjavac I don't know how to collapse info in the terminal.
Is there a similar example?

**Comment by zhmushan** [comment]
@justjavac I don't know how to collapse info in the terminal.
Is there a similar example?

File: js/console.ts (hunk 2 above)
Comment (zhmushan) [reply js/console.ts]: @justjavac please review.

**Comment by ry** [comment]
Waiting for #1355 to land. 

@kt3k Do you have some opinion on if we should use console.group or console. groupCollapsed printing to the same line?

**Comment by kt3k** [comment]
@ry 
I think console.groupCollapsed would be better for printing to the same line because it seems 'collapsed' and it doesn't have any use case in terminal (because there is no acutual  'collapse' behavior there).

**Comment by kt3k** [comment]
This PR has been landed as part of #1452.

**Comment by zhmushan** [comment]
> This PR has been landed as part of #1452.

👌

**Pull request closed without merging**